feat(pie-modal): Add modal component shell (#439)
justeattakeaway/pie: packages/components/pie-modal/README.md

<p align="center">
  <img align="center" src="../../../readme_image.png" height="200" alt="">
</p>

<p align="center">
  <a href="https://www.npmjs.com/@justeattakeaway/pie-modal">
    <img alt="GitHub Workflow Status" src="https://img.shields.io/npm/v/@justeattakeaway/pie-modal.svg">
  </a>

> </p>

Reviewer: Is the above intentional?

Author: This is adapted from [pie-button README file](https://github.com/justeattakeaway/pie/blob/main/packages/components/pie-button/README.md)

# pie-modal

This modal is a Web Component built using Lit.

## Local development

Install dependencies at the root
```
yarn
```

Navigate to this folder, compile with TypeScript and build with Vite
```
cd packages/components/pie-modal
yarn build
```

> compile using Vite (auto-compiles `dist` on save)

Reviewer 2: I'd argue that everyone knows what "Build and watch" means.

Would leave the technology used out of the readme.

Missing capital letter.

Reviewer 3: ```suggestion
Compile and watch for changes (auto-compiles `dist` on save)
```

```
yarn watch
```

> Local dev server using Vite (with hot module reloading)

Reviewer 3: ```suggestion
Run local dev server (with hot module reloading)
```
